src/ess/sans/contrib.py

```diff
@@ -0,0 +1,80 @@
+# Generic helpers that may end up as contributions to scipp if cleaned up
+import scipp as sc
+
+
+def midpoints(var, dim):
+    """
+    Utility function for finding middle points of variable
+    """
+    return 0.5 * (var[dim, 1:] + var[dim, :-1])
+
+
+def to_bin_centers(d, dim):
+    """
+    Utility function for setting centers of bins
+    """
+    edges = d.coords[dim].copy()
+    del d.coords[dim]
+    d.coords[dim] = 0.5 * (edges[dim, 1:] + edges[dim, :-1])
+
+
+def to_bin_edges(d, dim):
+    """
+    Utility function for setting bin edges
+    """
+    centers = d.coords[dim].copy()
```

Comment:
```suggestion
    centers = d.coords[dim]
```
The copy is not necessary for any scipp>=0.7